File: Graph/Graph/Minimum Number of Vertices to Reach All Nodes/Minimum Number of Vertices to Reach All Nodes.cpp

class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
       vector<int> indegrees(n, 0);
       vector<int> res;
       for(int i=0; i<edges.size(); i++)
            indegrees[edges[i][1]]++; 
        for(int i=0; i < n; i++)
            if(indegrees[i] == 0)
                res.push_back(i);
        return res;
    }
};

File: Graph/Graph/Minimum Number of Vertices to Reach All Nodes/Minimum Number of Vertices to Reach All Nodes.java
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n, List<List<Integer>> edges) {
       Map<Integer, Integer> indegree = new HashMap(); 
        for(List<Integer> edge: edges){
            int end = edge.get(1);
            indegree.put(end, indegree.getOrDefault(0, end) + 1 );
            
        }
        List<Integer> res = new ArrayList<>();
        for(int i = 0; i < n; i++){
           if(!indegree.containsKey(i)){
               res.add(i);
           } 
        }
        return res;
    }
}
